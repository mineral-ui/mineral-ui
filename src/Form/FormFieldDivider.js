/* @flow */
import React from 'react';
import { createStyledComponent } from '../styles';

type Props = Object;

export const componentTheme = (baseTheme: Object) => ({
  FormFieldDivider_borderColor: baseTheme.borderColor,
  FormFieldDivider_borderWidth: '1px',
  FormFieldDivider_margin: baseTheme.space_stack_sm,

  ...baseTheme
});

const Root = createStyledComponent(
  'div',
  ({ theme: baseTheme }) => {
    const theme = componentTheme(baseTheme);

    return {
      backgroundColor: theme.FormFieldDivider_borderColor,
      height: theme.FormFieldDivider_borderWidth,
      margin: `${theme.FormFieldDivider_margin} 0`
    };
  },
  {
    displayName: 'FormFieldDivider'
  }
);

/**
 * FormFieldDivider separates [FormFields](../form-field) to group form inputs.
 *
 * FormFieldDividers help your users grok forms with several inputs by hinting
 * at related fields, without explicitly adding a legend.
 */
export default function FormFieldDivider(props: Props) {
  return <Root {...props} role="separator" />;
}
