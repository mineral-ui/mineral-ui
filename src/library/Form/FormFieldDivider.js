/* @flow */
import React from 'react';
import { FormFieldDividerRoot as Root } from './styled';

import { formFieldDividerPropTypes } from './propTypes';
import type { FormFieldDividerProps } from './types';

const FormFieldDivider = (props: FormFieldDividerProps) => (
  <Root {...props} role="separator" />
);

FormFieldDivider.propTypes = formFieldDividerPropTypes;

export default FormFieldDivider;
