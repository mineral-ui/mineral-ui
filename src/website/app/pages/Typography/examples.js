/* @flow */
import React from 'react';
import { createStyledComponent } from '../../../../library/styles';
import Text from '../../../../library/Text';

type TypeExample = {
  description: string,
  examples: Array<Object>
};

type TypeExamples = Array<TypeExample>;

const styles = {
  list: ({ theme }) => ({
    fontSize: theme.fontSize_ui,
    listStyle: 'square',
    paddingLeft: theme.space_inline_lg
  })
};

const List = createStyledComponent('ul', styles.list);
const examples: TypeExamples = [
  {
    description: `### Long-form Text

Default text and caption styles are intended for sections of content that
provide description and assistance. **Paragraph text** is primarily used for
content sections. **Caption Text** is used to highlight or offset content, such
as inline help or when captioning dashboard content.`,
    examples: [
      {
        content: <Text>Paragraph</Text>,
        value: '<p>, <li>',
        color: 'color_gray_100',
        font: 'Regular (14/0.875)'
      },
      {
        content: <Text appearance="prose">Paragraph (Prose)</Text>,
        value: '<p.prose>',
        color: 'color_gray_100',
        font: 'Regular (16/1)'
      },
      {
        content: <Text appearance="mouse">Paragraph (Mouse)</Text>,
        value: '<p.mouse>',
        color: 'color_gray_100',
        font: 'Regular (11/0.6875)'
      },
      {
        content: (
          <List>
            <li>Alpha</li>
            <li>Bravo</li>
            <li>Charlie</li>
          </List>
        ),
        // the design notes have <ol> here as well,
        // but list-style will override any numbers shown on <ol> list items
        value: '<ul>, <ol>',
        color: 'color_gray_100',
        font: 'Regular (14/0.875)'
      },
      {
        content: <Text element="h5">Caption - Strong</Text>,
        value: '<h5>',
        color: 'color_gray_100',
        font: 'Bold (14/0.875)'
      },
      {
        content: <Text element="h6">Caption - Light</Text>,
        value: '<h6>',
        color: 'color_gray_80',
        font: 'Regular (14/0.875)'
      }
    ]
  },
  {
    description: `### Display Text

Use **Display text** sparingly to make a bold statement, or draw attention to
different sections in an application view. Display text is for short titles and
section divisions—it’s not for long-form content.`,
    examples: [
      {
        content: <Text element="h1">Page Title</Text>,
        value: '<h1>',
        color: 'color_gray_100',
        font: 'ExtraBold (34/2.125)'
      },
      {
        content: <Text element="h2">Large Section Header</Text>,
        value: '<h2>',
        color: 'color_gray_80',
        font: 'Bold (28/1.75)'
      },
      {
        content: <Text element="h3">Medium Section Header</Text>,
        value: '<h3>',
        color: 'color_gray_80',
        font: 'Bold (22/1.375)'
      },
      {
        content: <Text element="h4">Small Section Header</Text>,
        value: '<h4>',
        color: 'color_gray_80',
        font: 'Bold (18/1.125)'
      }
    ]
  }
];

export default examples;
