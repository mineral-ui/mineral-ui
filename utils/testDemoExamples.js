/* @flow */
import React from 'react';
import { mount } from 'enzyme';
import { LiveProvider, LivePreview } from 'react-live';
import ThemeProvider from '../src/library/themes/ThemeProvider';

type Example = {
  title: React$Node,
  scope?: Object,
  id: string,
  source?: string
};
type Examples = Array<Example>;
type Options = {
  exclude?: Array<string> // Example id
};

export default function testDemoExamples(
  examples: Examples,
  options: Options = {}
) {
  if (options.exclude) {
    const exclusions = options.exclude || [];
    examples = examples.filter(example => !exclusions.includes(example.id));
  }

  return describe('demo examples', () => {
    examples.filter(({ scope, source }) => scope && source).map(example => {
      it(example.id, () => {
        const component = mount(
          <ThemeProvider>
            <LiveProvider
              code={example.source}
              scope={example.scope}
              mountStylesheet={false}>
              <LivePreview />
            </LiveProvider>
          </ThemeProvider>
        );
        expect(component).toMatchSnapshot();
      });
    });
  });
}
