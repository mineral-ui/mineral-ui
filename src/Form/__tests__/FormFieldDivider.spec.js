/**
 * Copyright 2017 CA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import FormFieldDivider from '../FormFieldDivider';
import examples from '../../website/app/demos/Form/examples/form-field-divider';
import testDemoExamples from '../../../utils/testDemoExamples';

function shallowFormFieldDivider(props = {}) {
  const formFieldDividerProps = {
    ...props
  };

  return shallow(<FormFieldDivider {...formFieldDividerProps} />);
}

describe('FormFieldDivider', () => {
  describe('renders', () => {
    it('root', () => {
      const formFieldDivider = shallowFormFieldDivider();

      expect(formFieldDivider.exists()).toEqual(true);
    });
  });

  testDemoExamples(examples);
});
