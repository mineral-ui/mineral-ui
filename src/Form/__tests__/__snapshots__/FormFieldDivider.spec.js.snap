// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldDivider demo examples basic 1`] = `
.glamor-3,
[data-glamor-3] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
}

.glamor-3 *,
[data-glamor-3] *,
.glamor-3 *::before,
[data-glamor-3] *::before,
.glamor-3 *::after,
[data-glamor-3] *::after {
  box-sizing: inherit;
}

.glamor-0,
[data-glamor-0] {
  color: #58606e;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  font-size: 0.875em;
  font-weight: 600;
  justify-content: space-between;
  margin-bottom: 0.5em;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.glamor-0 > *,
[data-glamor-0] > * {
  align-self: flex-end;
  display: inline-block;
  -webkit-align-self: flex-end;
}

.glamor-2,
[data-glamor-2] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  align-items: center;
  cursor: text;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  height: 2.5em;
  position: relative;
  width: 100%;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-2 *,
[data-glamor-2] *,
.glamor-2 *::before,
[data-glamor-2] *::before,
.glamor-2 *::after,
[data-glamor-2] *::after {
  box-sizing: inherit;
}

.glamor-2 [role="img"],
[data-glamor-2] [role="img"] {
  display: block;
  fill: #c8d1e0;
  flex: 0 0 auto;
  margin: 0 0.5em;
  -webkit-flex: 0 0 auto;
}

.glamor-2 [role="img"]:last-of-type,
[data-glamor-2] [role="img"]:last-of-type {
  fill: #c8d1e0;
}

.glamor-1,
[data-glamor-1] {
  align-self: stretch;
  background-color: transparent;
  border: 0;
  box-shadow: none;
  color: #58606e;
  flex: 1 1 auto;
  font-family: inherit;
  font-size: 0.875em;
  min-width: 0;
  outline: 0;
  padding-left: 1.1428571428571428em;
  padding-right: 1.1428571428571428em;
  width: 100%;
  -webkit-align-self: stretch;
  -webkit-flex: 1 1 auto;
}

.glamor-1::placeholder,
[data-glamor-1]::placeholder,
.glamor-1[data-simulate-placeholder],
[data-glamor-1][data-simulate-placeholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-1::-ms-input-placeholder,
[data-glamor-1]::-ms-input-placeholder,
.glamor-1[data-simulate-msinputplaceholder],
[data-glamor-1][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-1:-ms-input-placeholder,
[data-glamor-1]:-ms-input-placeholder,
.glamor-1[data-simulate-msinputplaceholder],
[data-glamor-1][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-1::-ms-clear,
[data-glamor-1]::-ms-clear,
.glamor-1[data-simulate-msclear],
[data-glamor-1][data-simulate-msclear] {
  display: none;
}

.glamor-1:hover ~ div,
[data-glamor-1]:hover ~ div,
.glamor-1[data-simulate-hover] ~ div,
[data-glamor-1][data-simulate-hover] ~ div {
  border-color: #2e6fd9;
}

.glamor-1:focus ~ div,
[data-glamor-1]:focus ~ div,
.glamor-1[data-simulate-focus] ~ div,
[data-glamor-1][data-simulate-focus] ~ div {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.glamor-1:active ~ div,
[data-glamor-1]:active ~ div,
.glamor-1[data-simulate-active] ~ div,
[data-glamor-1][data-simulate-active] ~ div {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.glamor-1 ~ div,
[data-glamor-1] ~ div {
  background-color: #fff;
  border-color: #c8d1e0;
  border-radius: 0.1875em;
  border-style: solid;
  border-width: 1px;
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}

.glamor-4,
[data-glamor-4] {
  background-color: #c8d1e0;
  height: 1px;
  margin: 0.5em 0;
}

<ThemeProvider>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor_danger": "#db2929",
        "backgroundColor_danger_active": "#c71818",
        "backgroundColor_danger_activeMuted": "#f79999",
        "backgroundColor_danger_focus": "#db2929",
        "backgroundColor_danger_hover": "#eb4d4d",
        "backgroundColor_disabled": "#ebeff5",
        "backgroundColor_input": "#fff",
        "backgroundColor_link_focus": "#dde3ed",
        "backgroundColor_success": "#0a8f4c",
        "backgroundColor_success_active": "#06783f",
        "backgroundColor_success_activeMuted": "#95f5c3",
        "backgroundColor_success_focus": "#0a8f4c",
        "backgroundColor_success_hover": "#10a35a",
        "backgroundColor_warning": "#e05b2b",
        "backgroundColor_warning_active": "#cf4615",
        "backgroundColor_warning_activeMuted": "#fab69d",
        "backgroundColor_warning_focus": "#e05b2b",
        "backgroundColor_warning_hover": "#ed774c",
        "borderColor": "#c8d1e0",
        "borderColor_active": "#2e6fd9",
        "borderColor_danger": "#db2929",
        "borderColor_danger_active": "#c71818",
        "borderColor_danger_focus": "#c71818",
        "borderColor_danger_hover": "#c71818",
        "borderColor_focus": "#2e6fd9",
        "borderColor_hover": "#2e6fd9",
        "borderColor_success": "#10a35a",
        "borderColor_success_active": "#0a8f4c",
        "borderColor_success_focus": "#0a8f4c",
        "borderColor_success_hover": "#0a8f4c",
        "borderColor_warning": "#e05b2b",
        "borderColor_warning_active": "#cf4615",
        "borderColor_warning_focus": "#cf4615",
        "borderColor_warning_hover": "#cf4615",
        "borderRadius_1": "0.1875em",
        "color_black": "#1d1f24",
        "color_caption": "#58606e",
        "color_gray_10": "#f5f7fa",
        "color_gray_100": "#333840",
        "color_gray_20": "#ebeff5",
        "color_gray_30": "#dde3ed",
        "color_gray_40": "#c8d1e0",
        "color_gray_50": "#afbacc",
        "color_gray_60": "#8e99ab",
        "color_gray_70": "#707a8a",
        "color_gray_80": "#58606e",
        "color_gray_90": "#434a54",
        "color_placeholder": "#8e99ab",
        "color_text": "#333840",
        "color_text_danger": "#db2929",
        "color_text_danger_active": "#c71818",
        "color_text_danger_focus": "#db2929",
        "color_text_danger_hover": "#eb4d4d",
        "color_text_disabled": "#afbacc",
        "color_text_ondanger": "#fff",
        "color_text_onprimary": "#fff",
        "color_text_onsuccess": "#fff",
        "color_text_onwarning": "#fff",
        "color_text_primary": "#2e6fd9",
        "color_text_primary_active": "#1f5dc2",
        "color_text_primary_focus": "#2e6fd9",
        "color_text_primary_hover": "#4a89e8",
        "color_text_success": "#06783f",
        "color_text_success_active": "#046132",
        "color_text_success_focus": "#06783f",
        "color_text_success_hover": "#0a8f4c",
        "color_text_warning": "#cf4615",
        "color_text_warning_active": "#bd3909",
        "color_text_warning_focus": "#cf4615",
        "color_text_warning_hover": "#e05b2b",
        "color_theme_10": "#e6eefc",
        "color_theme_100": "#0f397d",
        "color_theme_20": "#c2dbfc",
        "color_theme_30": "#9dc2fa",
        "color_theme_40": "#72a5f2",
        "color_theme_50": "#4a89e8",
        "color_theme_60": "#2e6fd9",
        "color_theme_70": "#1f5dc2",
        "color_theme_80": "#164ea8",
        "color_theme_90": "#114091",
        "color_white": "#fff",
        "direction": "ltr",
        "fontFamily": "Open Sans",
        "fontFamily_monospace": "\\"SF Mono\\", \\"Droid Sans Mono\\", \\"Source Code Pro\\", Monaco, Consolas, \\"Courier New\\", Courier, monospace",
        "fontFamily_system": "-apple-system, system-ui, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"",
        "fontSize_base": 16,
        "fontSize_h1": "2.125em",
        "fontSize_h2": "1.75em",
        "fontSize_h3": "1.375em",
        "fontSize_h4": "1.125em",
        "fontSize_h5": "0.875em",
        "fontSize_h6": "0.875em",
        "fontSize_mouse": "0.6875em",
        "fontSize_prose": "1em",
        "fontSize_ui": "0.875em",
        "fontWeight_bold": 700,
        "fontWeight_extraBold": 800,
        "fontWeight_regular": 400,
        "fontWeight_semiBold": 600,
        "lineHeight": 1.25,
        "lineHeight_prose": 1.5,
        "shadow_1": "0 1px 2px 0 rgba(0,0,0,0.4)",
        "shadow_2": "0 2px 4px 0 rgba(0,0,0,0.2), 0 1px 2px 0 rgba(0,0,0,0.4)",
        "shadow_3": "0 4px 8px 0 rgba(0,0,0,0.2), 0 2px 4px 0 rgba(0,0,0,0.4)",
        "shadow_4": "0 8px 16px 0 rgba(0,0,0,0.2), 0 4px 8px 0 rgba(0,0,0,0.4)",
        "shadow_5": "0 24px 24px -8px rgba(0,0,0,0.4), 0 8px 16px 0 rgba(0,0,0,0.4)",
        "size_jumbo": "3.25em",
        "size_large": "2.5em",
        "size_medium": "2em",
        "size_small": "1.5em",
        "space_inline_lg": "1.5em",
        "space_inline_md": "1em",
        "space_inline_sm": "0.5em",
        "space_inline_xl": "2em",
        "space_inline_xs": "0.25em",
        "space_inline_xxl": "4em",
        "space_inline_xxs": "0.125em",
        "space_inset_lg": "1.5em",
        "space_inset_md": "1em",
        "space_inset_sm": "0.5em",
        "space_stack_lg": "1.5em",
        "space_stack_md": "1em",
        "space_stack_sm": "0.5em",
        "space_stack_xl": "2em",
        "space_stack_xs": "0.25em",
        "space_stack_xxl": "4em",
        "space_stack_xxs": "0.125em",
        "zIndex_100": 100,
        "zIndex_1600": 1600,
        "zIndex_200": 200,
        "zIndex_400": 400,
        "zIndex_800": 800,
      }
    }
  >
    <LiveProvider
      code="
    <DemoLayout>
      <FormField label=\\"Old Password\\">
        <TextInput type=\\"password\\" />
      </FormField>

      <FormFieldDivider />

      <FormField label=\\"New Password\\">
        <TextInput type=\\"password\\" />
      </FormField>
      <FormField label=\\"Repeat New Password\\">
        <TextInput type=\\"password\\" />
      </FormField>
    </DemoLayout>
  "
      mountStylesheet={false}
      noInline={false}
      scope={
        Object {
          "DemoLayout": [Function],
          "FormField": [Function],
          "FormFieldDivider": [Function],
          "TextInput": [Function],
        }
      }
    >
      <div
        className="react-live"
        scope={
          Object {
            "DemoLayout": [Function],
            "FormField": [Function],
            "FormFieldDivider": [Function],
            "TextInput": [Function],
          }
        }
      >
        <LivePreview>
          <div
            className="react-live-preview"
          >
            <glamorous(div)>
              <div
                className="glamor-13"
              >
                <FormField
                  label="Old Password"
                  requiredText="Required"
                >
                  <FormField>
                    <div
                      className="glamor-3"
                    >
                      <label>
                        <glamorous(div)
                          hideLabel={undefined}
                        >
                          <div
                            className="glamor-0"
                          >
                            Old Password
                          </div>
                        </glamorous(div)>
                        <TextInput
                          aria-describedby={undefined}
                          key=".0"
                          required={undefined}
                          type="password"
                          variant={undefined}
                        >
                          <TextInput
                            className={undefined}
                            size="large"
                            variant={undefined}
                          >
                            <div
                              className="glamor-2"
                            >
                              <glamorous(input)
                                aria-describedby={undefined}
                                aria-invalid={undefined}
                                aria-required={undefined}
                                disabled={undefined}
                                iconEnd={undefined}
                                iconStart={undefined}
                                innerRef={[Function]}
                                prefix={undefined}
                                readOnly={undefined}
                                required={undefined}
                                size="large"
                                suffix={undefined}
                                type="password"
                                variant={undefined}
                              >
                                <input
                                  aria-describedby={undefined}
                                  aria-invalid={undefined}
                                  aria-required={undefined}
                                  className="glamor-1"
                                  disabled={undefined}
                                  prefix={undefined}
                                  readOnly={undefined}
                                  required={undefined}
                                  size="large"
                                  type="password"
                                />
                              </glamorous(input)>
                              <div />
                            </div>
                          </TextInput>
                        </TextInput>
                      </label>
                    </div>
                  </FormField>
                </FormField>
                <FormFieldDivider>
                  <FormFieldDivider
                    role="separator"
                  >
                    <div
                      className="glamor-4"
                      role="separator"
                    />
                  </FormFieldDivider>
                </FormFieldDivider>
                <FormField
                  label="New Password"
                  requiredText="Required"
                >
                  <FormField>
                    <div
                      className="glamor-3"
                    >
                      <label>
                        <glamorous(div)
                          hideLabel={undefined}
                        >
                          <div
                            className="glamor-0"
                          >
                            New Password
                          </div>
                        </glamorous(div)>
                        <TextInput
                          aria-describedby={undefined}
                          key=".0"
                          required={undefined}
                          type="password"
                          variant={undefined}
                        >
                          <TextInput
                            className={undefined}
                            size="large"
                            variant={undefined}
                          >
                            <div
                              className="glamor-2"
                            >
                              <glamorous(input)
                                aria-describedby={undefined}
                                aria-invalid={undefined}
                                aria-required={undefined}
                                disabled={undefined}
                                iconEnd={undefined}
                                iconStart={undefined}
                                innerRef={[Function]}
                                prefix={undefined}
                                readOnly={undefined}
                                required={undefined}
                                size="large"
                                suffix={undefined}
                                type="password"
                                variant={undefined}
                              >
                                <input
                                  aria-describedby={undefined}
                                  aria-invalid={undefined}
                                  aria-required={undefined}
                                  className="glamor-1"
                                  disabled={undefined}
                                  prefix={undefined}
                                  readOnly={undefined}
                                  required={undefined}
                                  size="large"
                                  type="password"
                                />
                              </glamorous(input)>
                              <div />
                            </div>
                          </TextInput>
                        </TextInput>
                      </label>
                    </div>
                  </FormField>
                </FormField>
                <FormField
                  label="Repeat New Password"
                  requiredText="Required"
                >
                  <FormField>
                    <div
                      className="glamor-3"
                    >
                      <label>
                        <glamorous(div)
                          hideLabel={undefined}
                        >
                          <div
                            className="glamor-0"
                          >
                            Repeat New Password
                          </div>
                        </glamorous(div)>
                        <TextInput
                          aria-describedby={undefined}
                          key=".0"
                          required={undefined}
                          type="password"
                          variant={undefined}
                        >
                          <TextInput
                            className={undefined}
                            size="large"
                            variant={undefined}
                          >
                            <div
                              className="glamor-2"
                            >
                              <glamorous(input)
                                aria-describedby={undefined}
                                aria-invalid={undefined}
                                aria-required={undefined}
                                disabled={undefined}
                                iconEnd={undefined}
                                iconStart={undefined}
                                innerRef={[Function]}
                                prefix={undefined}
                                readOnly={undefined}
                                required={undefined}
                                size="large"
                                suffix={undefined}
                                type="password"
                                variant={undefined}
                              >
                                <input
                                  aria-describedby={undefined}
                                  aria-invalid={undefined}
                                  aria-required={undefined}
                                  className="glamor-1"
                                  disabled={undefined}
                                  prefix={undefined}
                                  readOnly={undefined}
                                  required={undefined}
                                  size="large"
                                  type="password"
                                />
                              </glamorous(input)>
                              <div />
                            </div>
                          </TextInput>
                        </TextInput>
                      </label>
                    </div>
                  </FormField>
                </FormField>
              </div>
            </glamorous(div)>
          </div>
        </LivePreview>
      </div>
    </LiveProvider>
  </ThemeProvider>
</ThemeProvider>
`;
